File: src/main/java/net/hectus/neobb/turn/card_game/CTJackOLantern.java
package net.hectus.neobb.turn.card_game;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.util.MinecraftTime;
import org.bukkit.Material;
import org.bukkit.block.Block;
import org.bukkit.inventory.ItemStack;

public class CTJackOLantern extends CardTurn {
    public CTJackOLantern(NeoPlayer player) { super(player); }
    public CTJackOLantern(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public double damage() {
        return player.game.time() == MinecraftTime.MIDNIGHT ? 5 : 2;
    }

    @Override
    public ItemStack item() {
        return new ItemStack(Material.JACK_O_LANTERN);
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/card_game/CTJackOLantern.kt
package net.hectus.neobb.modes.turn.card_game

import com.marcpg.libpg.storing.Cord
import com.marcpg.libpg.util.MinecraftTime
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.Material
import org.bukkit.block.Block
import org.bukkit.inventory.ItemStack

class CTJackOLantern(data: Block?, cord: Cord?, player: NeoPlayer?) : CardTurn(data, cord, player) {
    override val damage: Double
        get() = if (player?.game?.time == MinecraftTime.MIDNIGHT) 5.0 else 2.0

    override fun item(): ItemStack = ItemStack(Material.JACK_O_LANTERN)
}
